packages/explorer/src/app/interact/FunctionField.tsx

```diff
@@ -0,0 +1,194 @@
+      } catch (error: Error | unknown) {
+        console.log("error:", error);
+
+        let msg = "Something went wrong. Please try again.";
+        if (error instanceof Error) {
+          msg = error.message;
+        }
+
+        toast.error(msg, {
+          id: toastId,
+        });
+      } finally {
+        console.log(account);
+        queryClient.invalidateQueries({
+          queryKey: [
+            "balance",
+            {
+              address: account,
+              chainId,
+            },
+          ],
+        });
+      }
+    }
+  }
+
+  const inputsLabel = abi?.inputs.map((input) => input.type).join(", ");
+  return (
+    <Form {...form}>
+      <form
+        onSubmit={form.handleSubmit(onSubmit)}
+        id={abi.name}
+        className="space-y-4 pb-4"
+      >
+        <h3 className="pt-4 font-semibold">
+          <span className="text-orange-500">{abi?.name}</span>
+          <span className="opacity-50">
+            {inputsLabel && ` (${inputsLabel})`}
+          </span>
+          <span className="ml-2 opacity-50">
+            {abi.stateMutability === "payable" && (
+              <Coins className="mr-2 inline-block h-4 w-4" />
+            )}
+            {(abi.stateMutability === "view" ||
+              abi.stateMutability === "pure") && (
+              <Eye className="mr-2 inline-block h-4 w-4" />
+            )}
+            {abi.stateMutability === "nonpayable" && (
+              <Send className="mr-2 inline-block h-4 w-4" />
+            )}
+          </span>
+        </h3>
+
+        {abi?.inputs.map((input, idx) => {
+          return (
+            <FormField
+              key={idx}
+              control={form.control}
+              name={`inputs.${idx}`}
+              render={({ field }) => (
+                <FormItem>
+                  <FormLabel>{input.name}</FormLabel>
+                  <FormControl>
+                    <Input placeholder={input.type} {...field} />
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+          );
+        })}
+
+        {abi.stateMutability === "payable" && (
+          <FormField
+            control={form.control}
+            name="value"
+            render={({ field }) => (
+              <FormItem>
+                <FormLabel>ETH value</FormLabel>
+                <FormControl>
+                  <Input placeholder="uint256" {...field} />
+                </FormControl>
+                <FormMessage />
+              </FormItem>
+            )}
+          />
+        )}
+
+        <Button type="submit">
+          {(abi.stateMutability === "view" || abi.stateMutability === "pure") &&
+            "Read"}
+          {(abi.stateMutability === "payable" ||
+            abi.stateMutability === "nonpayable") &&
+            "Write"}
+        </Button>
+
+        {result && (
+          <pre className="text-md rounded border p-3 text-sm">{result}</pre>
+        )}
```

Comment:
I personally have found this pattern to be too much of a foot-gun and now use the following in all my React ternaries:
```suggestion
        {result ? (
          <pre className="text-md rounded border p-3 text-sm">{result}</pre>
        ) : null}
```

the reason is if `result` is ever `0` (zero), it will be rendered as `0` in the UI rather than nothing

although this may not be a concern now or in this context, in some cases we may refactor a variable used in rendering with the possibility of being zero and then suddenly have weird rendering bugs in UI that could be hard to spot or track down

Reply:
Understood, probably result is not going be 0 in this case but down the line it could change so yeah, makes sense, updated.